edxval/tests/test_api.py

```diff
@@ -436,3 +622,74 @@ def test_get_info_queries_for_only_video(self):
         with self.assertNumQueries(4):
             api.get_video_info(constants.VIDEO_DICT_ZEBRA.get("edx_video_id"))
 
+
+class TestCopyCourse(TestCase):
+    """Tests rerun course"""
+
+    def setUp(self):
+        """
+        Creates a course for testing
+
+        Creates two videos, each with two encodings.
+        """
+        Profile.objects.create(**constants.PROFILE_DICT_MOBILE)
+        Profile.objects.create(**constants.PROFILE_DICT_DESKTOP)
+        # 1st video
+        video = Video.objects.create(**constants.VIDEO_DICT_FISH)
+        EncodedVideo.objects.create(
+            video=video,
+            profile=Profile.objects.get(profile_name="mobile"),
+            **constants.ENCODED_VIDEO_DICT_MOBILE
+        )
+        EncodedVideo.objects.create(
+            video=video,
+            profile=Profile.objects.get(profile_name="desktop"),
+            **constants.ENCODED_VIDEO_DICT_DESKTOP
+        )
+        self.course_id = 'test-course'
+        CourseVideo.objects.create(video=video, course_id=self.course_id)
+        # 2nd video
+        video = Video.objects.create(**constants.VIDEO_DICT_STAR)
+        EncodedVideo.objects.create(
+            video=video,
+            profile=Profile.objects.get(profile_name="mobile"),
+            **constants.ENCODED_VIDEO_DICT_MOBILE2
+        )
+        EncodedVideo.objects.create(
+            video=video,
+            profile=Profile.objects.get(profile_name="desktop"),
+            **constants.ENCODED_VIDEO_DICT_DESKTOP2
+        )
+        CourseVideo.objects.create(video=video, course_id=self.course_id)
+
+    def _check_video_course_list(self, video, expected_course_list):
+        """
+        Verifies that given Video has the courses in the course list
+        """
+        courses = VideoSerializer(video).data['courses']
+        self.assertEqual(sorted(courses), sorted(expected_course_list))
+
+    def test_successful_rerun(self):
+        """Tests a successful rerun and 2nd rerun"""
+        api.copy_course_videos('test-course', 'meow-rerun')
+        videos = Video.objects.all()
+        for video in videos:
+            self._check_video_course_list(video,
+                                          ['test-course',
+                                           'meow-rerun'])
+
+        api.copy_course_videos('meow-rerun', 'woof-bark')
+        videos = Video.objects.all()
+        for video in videos:
+            self._check_video_course_list(video,
+                                          ['test-course',
+                                           'meow-rerun',
+                                           'woof-bark'])
+
+    def test_same_course_ids(self):
+        """Tests when the new course id name is the same as the original"""
+        api.copy_course_videos('test-course', 'test-course')
```

Comment:
need to add assert here to make sure the list of courses for the video remains the same.